plugin/clang.vim

```diff
@@ -911,16 +935,18 @@ endf
 "       'stderr':  // updated in async mode
 "     }
```

Comment:
Use `ClangExecuteDone('', '')` is strange and `has('nvim')` should be used as less as possible, it is not good for a project.
You can add a new function to split read result and trigger completion, now you can use `s:ClangExecuteDoneTriggerCompletion` to trigger completion without overwriting stdout and stderr

```viml
" {{{ ClangExecuteDone
" Called by vim-client when clang is returned in asynchronized mode.
"
" Buffer vars:
"     b:clang_state => {
"       'state' :  // updated to 'sync' in async mode
"       'stdout':  // updated in async mode
"       'stderr':  // updated in async mode
"     }
func! ClangExecuteDone(tmp1, tmp2)
  let l:res = s:DeleteAfterReadTmps([a:tmp1, a:tmp2])
  let b:clang_state['stdout'] = l:res[0]
  let b:clang_state['stderr'] = l:res[1]
  call s:ClangExecuteDoneTriggerCompletion()
endf
" }}}
" {{{ s:ClangExecuteDoneTriggerCompletion
" Won't overwirte 'stdout' and 'stderr' in b:clang_state
" Buffer vars:
"     b:clang_state => {
"       'state' :  // updated to 'sync' in async mode
"       'stdout':  // updated in async mode
"       'stderr':  // updated in async mode
"     }
func! s:ClangExecuteDoneTriggerCompletion()
  let b:clang_state['state'] = 'sync'
  call s:PDebug("ClangExecuteDoneTriggerCompletion::stdout", b:clang_state['stdout'], 3)
  call s:PDebug("ClangExecuteDoneTriggerCompletion::stderr", b:clang_state['stderr'], 2)
  call feedkeys("\<Esc>a")
  " As the default action of <C-x><C-o> causes a 'pattern not found'
  " when the result is empty, which break our input, that's really painful...
  if ! empty(l:res[0])
    call feedkeys("\<C-x>\<C-o>")
  else
    call ClangComplete(0, ClangComplete(1, 0))
  endif
endf
"}}}
```